plugins/sep/sli/slipresentation.cc

```diff
@@ -303,17 +305,25 @@ SliPresentation::getPixelAverages(Scroom::Utils::Rectangle<int> area) {
     B = surfaceBegin[offset + 0];
     G = surfaceBegin[offset + 1];
     R = surfaceBegin[offset + 2];
-    // A = surfaceBegin[offset+3]; // don't need this
+    A = surfaceBegin[offset + 3];
 
     c = (255.0 - R);
     m = (255.0 - G);
     y = (255.0 - B);
     k = std::min({c, m, y});
```

Comment:
This seems to be trying to convert the background color from RGB to CMYK. Going from 3 to 4 variables leaves some room for interpretation of how to distribute the contribution of the K-channel to the CMY-channels. Let's not go into using a color profile for these conversions, but that they make a straight-forward conversion from RGB to CMYK:

I believe this conversion is incorrect, as it does not match the conversions shown here:

https://www.codeproject.com/Articles/4488/XCmyk-CMYK-to-RGB-Calculator-with-source-code
https://stackoverflow.com/a/2426490/978020
http://www.rapidtables.com/convert/color/rgb-to-cmyk.htm

Additionally, the alpha-channel A is only checked for level 0 (fully transparent). Is this because the UI will only set transparent and non-transparent?

I would like to see this conversion as a utility function that converts RGB(A) into CMYK values. It may definitely be used elsewhere.

Reply:
If I remember correctly, I used the formula from this document: https://www.eembc.org/techlit/datasheets/cmyk_consumer.pdf
From my tests, this seems correct. The only difference is that the values are kept between 0 and 255, to keep it consistent with what the pipette tool expects, and not normalized to be between 0 and 1. 

Since we established that we were not going to support transparency, the Alpha fix is only relevant for certain cases involving SLI files with particular offsets. Since we start drawing with a completely blank (all 0s) surface, if no layer is drawn on a particular area, then the values for the pixels in that area will remain 0's (since we only draw/convert the area covered by the layers). Visually, this does not change anything and the default white background of Scroom will show through in this case. However, the pipette tool will show a value of 255 for K, instead of 0, because that white background is not really a layer (it's drawn separately) and the pipette will only have access to the transparent pixels. So this PR fixes that issue. 
Providing the bottom background as an SLI layer itself will avoid this issue altogether.

I can definitely move the conversion function into the SliHelpers file in a separate PR. Note that this formula can only be used for SLI at the moment, as the other plugins use the TiledBitmap instead.

Reply:
Ah, I see my misunderstanding. I misunderstood the definition of the lowercase c, m, y, k variables. They are calculating the (intermediate) complement of the RGB values. 

Thanks for explaning the transparency and the issue with the pipette.

For now, the color conversion can stay here, but it would be good to refactor it to a separate function if there is still time.